paparazzi/src/test/java/app/cash/paparazzi/helper/TestSnapshotVerifier.kt

```diff
@@ -0,0 +1,33 @@
+package app.cash.paparazzi.helper
+
+import app.cash.paparazzi.Snapshot
+import app.cash.paparazzi.SnapshotHandler
+import app.cash.paparazzi.internal.ImageUtils
+import java.awt.image.BufferedImage
+import java.io.File
+import javax.imageio.ImageIO
+
+internal class TestSnapshotVerifier(val failureDirProvider: () -> File) : SnapshotHandler {
```

Comment:
there is a reason. Since this is instantiated as a dependency on the Paparazzi test rule, it invokes `temporaryDirectory.newFolder()`. You can't invoke that until `TemporaryDirectory` test rule has started.

I decided to defer the failureDir since it is not needed till later.